src/sonic-host-services/host_modules/showtech.py

```diff
@@ -0,0 +1,55 @@
+"""Show techsupport command handler"""
+
+import host_service
+import subprocess
+import re
+
+MOD_NAME = 'showtech'
+
+class Showtech(host_service.HostModule):
+    """DBus endpoint that executes the "show techsupport" command
+    """
+    @host_service.method(host_service.bus_name(MOD_NAME), in_signature='s', out_signature='is')
+    def info(self, date):
+
+        ERROR_TAR_FAILED = 5
+        ERROR_PROCFS_SAVE_FAILED = 6
+        ERROR_INVALID_ARGUMENT = 10
+
+        err_dict = {ERROR_INVALID_ARGUMENT: 'Invalid input: Incorrect DateTime format',
+                    ERROR_TAR_FAILED: 'Failure saving information into compressed output file',
+                    ERROR_PROCFS_SAVE_FAILED: 'Saving of process information failed'}
+
+        cmd = ['/usr/local/bin/generate_dump']
+        if date:
+            cmd.append("-s")
+            cmd.append(date)
+
+        try:
+            rc = 0
+            output = subprocess.check_output(cmd)
```

Comment:
>check_output [](http://example.com/codeflow?start=32&length=12)

Use `encoding` option, so no need to `decode` below?

Reply:
Yes. Because the "check_output" call has been replaced with "run" specifying text encoding, the utf-8 decode is no longer needed and has been deleted.